README.md

```diff
@@ -2,92 +2,707 @@
+Create a `PelicanFileSystem` instance and provide it with your federation's discovery URL:
 
 ```python
 from pelicanfs.core import PelicanFileSystem
 
+# Connect to the OSDF federation
 pelfs = PelicanFileSystem("pelican://osg-htc.org")
+
+# List objects in a namespace
+objects = pelfs.ls('/ospool/uc-shared/public/OSG-Staff/')
+
+# Read an object
+content = pelfs.cat('/ospool/uc-shared/public/OSG-Staff/validation/test.txt')
+print(content)
 ```
 
-Once `pelfs` is pointed at your federation's director, fsspec commands can be applied to Pelican namespaces. For example:
+### Using the OSDF Scheme
+
+For convenience, you can use the `osdf://` scheme directly with fsspec or the `OSDFFileSystem` class:
 
 ```python
-hello_world = pelfs.cat('/ospool/uc-shared/public/OSG-Staff/validation/test.txt')
-print(hello_world)
+from pelicanfs.core import OSDFFileSystem
+import fsspec
+
+# Using OSDFFileSystem (automatically connects to osg-htc.org)
+osdf = OSDFFileSystem()
+objects = osdf.ls('/ospool/uc-shared/public/')
+
+# Or use fsspec directly with the osdf:// scheme
+with fsspec.open('osdf:///ospool/uc-shared/public/OSG-Staff/validation/test.txt', 'r') as f:
+    content = f.read()
 ```
 
-### Getting an FSMap
+## Object Operations
 
-Sometimes various systems that interact with an fsspec want a key-value mapper rather than a url. To do that, call the `PelicanMap` function with the namespace path and a `PelicanFileSystem` object rather than using the fsspec `get_mapper` call. For example:
+### Listing Objects and Collections
 
 ```python
-from pelicanfs.core import PelicanFileSystem, PelicanMap
+import fsspec
 
-pelfs = PelicanFileSystem("some-director-url")
-file1 = PelicanMap("/namespace/file/1", pelfs=pelfs)
-file2 = PelicanMap("/namespace/file/2", pelfs=pelfs)
-ds = xarray.open_mfdataset([file1,file2], engine='zarr')
+# Method 1: Using fsspec functions with schemes (recommended for most users)
+objects = fsspec.ls('osdf:///ospool/uc-shared/public/')
+
+# List with details (size, type, etc.)
+objects_detailed = fsspec.ls('osdf:///ospool/uc-shared/public/', detail=True)
+
+# Recursively find all objects
+all_objects = fsspec.find('osdf:///ospool/uc-shared/public/')
+
+# Find objects with depth limit
+objects = fsspec.find('osdf:///ospool/uc-shared/public/', maxdepth=2)
+
+# Method 2: Using PelicanFileSystem directly (for more control)
+from pelicanfs.core import PelicanFileSystem
+pelfs = PelicanFileSystem("pelican://osg-htc.org")
+objects = pelfs.ls('/ospool/uc-shared/public/')
 ```
 
-### Specifying Endpoints
+### Pattern Matching with Glob
+
+```python
+import fsspec
+
+# Method 1: Using fsspec.glob with schemes (recommended)
+csv_objects = fsspec.glob('osdf:///ospool/uc-shared/public/**/*.csv')
+
+# Find objects with depth limit
+json_objects = fsspec.glob('osdf:///ospool/uc-shared/public/**/*.json', maxdepth=3)
```

Comment:
As I understand, this can be a very expensive operation for large namespaces (not sure whether it's worse for wide or deep namespaces). Maybe give a warning about that.